### ISSUE TITLE
Wrong free() in Clipboard.c/nxagentCollectPropertyEvent?

### ISSUE THREAD

**Issue by uli42** [ArcticaProject/nx-libs#553]
```
   result = NXGetCollectedProperty(nxagentDisplay,
                                  resource,
                                  &atomReturnType,
                                  &resultFormat,
                                  &ulReturnItems,
                                  &ulReturnBytesLeft,
                                  &pszReturnData);

...
    
  if (pszReturnData != NULL)
    {
      XFree(pszReturnData);
    }

...
    if (pszReturnData != NULL)
    {
      XFree(pszReturnData);
    }
...
      
        if (pszReturnData != NULL)
        {
          free(pszReturnData);
        }
...
        if (pszReturnData != NULL)
        {
          free(pszReturnData);
        }#
...
  XFree(pszReturnData);
  pszReturnData = NULL;
}
```
pszReturnData is filled once at the beginning by NXGetCollectedProperty(). Some of the code in nxagentCollectPropertyEvent() calls XFree(pszReturnData), dome calls free(pszReturnData), so this is suspicious. Looks like acf87144d019f18e646501657d9082c6eba77f54 was incomplete, for whatever reason.

**Comment by uli42** [comment]
I think this should be fixed before making another 3.5.99 release

**Comment by sunweaver** [comment]
ok, I'll try to give it a shot tomorrow...
